### PR TITLE
Task_11980: Init Project.

### PR DESCRIPTION
**Pull request by khanhnn-1791**
### Issue Link :link:

https://edu-redmine.sun-asterisk.vn/issues/11980
### Goals :soccer:

- Initial project with MVVM structure
Design UI for Launch screen
 - Logo Image
 - Copyright Label

### Implementation Details :construction:

Screen Shot 2019-05-13 at 8 12 03 AM

### CODE REVIEW

**Comment by khanhnn-1791** [comment]
ready

File: MovieApp/Data/Source/Remote/API/Middleware/CustomRequestAdapter.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import Foundation
+import Alamofire
+
+final class CustomRequestAdapter: RequestAdapter {
+    private let userDefault = UserDefaults()
+    
+    func adapt(_ urlRequest: URLRequest) throws -> URLRequest {
+        var urlRequest = urlRequest
+        if let accessToken = userDefault.string(forKey: Constants.keyAccessToken) {
```
Comment (tienpx3010) [inline MovieApp/Data/Source/Remote/API/Middleware/CustomRequestAdapter.swift]:
```suggestion
        if let accessToken = UserDefaults.standard.string(forKey: Constants.keyAccessToken) {
```

File: MovieApp/Data/Source/Remote/API/Request/BaseUploadRequest.swift [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+//  MovieApp
+//
+//  Created by nguyen.nam.khanh on 5/10/19.
+//  Copyright © 2019 nguyen.nam.khanh. All rights reserved.
+//
+
+import Foundation
+import ObjectMapper
+
+class BaseUploadRequest: NSObject {
```
Comment (tienpx3010) [inline MovieApp/Data/Source/Remote/API/Request/BaseUploadRequest.swift]: App nếu không sử dụng upload file thì bỏ

File: MovieApp/Data/Source/Remote/Service/APIService.swift [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        print("link: %@", input.url)
+        print("body: %@", input.body ?? "No Body")
+        print("------------ END REQUEST INPUT\n")
+        
+        return Observable.create { observer in
+            self.alamofireManager.request(input.url,
+                                          method: input.requestType,
+                                          parameters: input.body,
+                                          encoding: input.encoding)
+                .validate(statusCode: 200..<500)
```
Comment (tienpx3010) [inline MovieApp/Data/Source/Remote/Service/APIService.swift]:
```suggestion
                .validate(statusCode: 200..<511)
```

File: MovieApp/Extension/Observable+Extension.swift [hunk 4]
```diff
@@ -0,0 +1,60 @@
+//
+//  Observable+Extension.swift
```
Comment (tienpx3010) [inline MovieApp/Extension/Observable+Extension.swift]:
```suggestion
//  Observable+.swift
```
```suggestion
//  Observable+Ext.swift
```

File: MovieApp/Extension/Observable+Extension.swift [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+extension ObserverType where E == Void {
+    public func onNext() {
+        onNext(())
+    }
+}
+
+extension Reactive where Base: SVProgressHUD {
+    public static var isAnimating: Binder<Bool> {
+        return Binder(UIApplication.shared) { _, isVisible in
```
Comment (tienpx3010) [inline MovieApp/Extension/Observable+Extension.swift]:
```suggestion
        return Binder(base) { _, isVisible in
```

**Comment by khanhnn-1791** [comment]
ready

**Pull request merged**